Resources/public/mapbender.element.digitizer.js

```diff
@@ -1285,22 +1265,40 @@
         /**
          * Copy feature
          *
-         * @param feature
+         * @param {OpenLayers.Feature} feature
          */
-        copyFeature: function(feature) {
+        copyFeature: function(feature ) {
             var widget = this;
             var schema = feature.schema;
+            var layer = feature.layer;
             var newFeature = feature.clone();
             var config = schema.copy;
             var defaultAttributes = getValueOrDefault(config, "data",{});
+            var allowCopy = true;
 
-            newFeature.attrubites = _.extend({}, defaultAttributes, feature.attributes);
-            _.each(defaultAttributes,function(key,value) {
-               if(newFeature.attrubites[key] === null){
-                   newFeature.attrubites[key] = value;
-               }
+            _.each(schema.copy.rules, function(ruleCode) {
+                var f = feature;
+                eval('allowCopy = ' + ruleCode + ';');
+                if(!allowCopy) {
+                    return false;
+                }
             });
-            newFeature.isNew = true;
+
+            if(!allowCopy) {
+                $.notify(translate('feature.clone.on.error'));
+                return;
+            }
+
+            var newAttributes = {};
+            _.extend(newAttributes, defaultAttributes);
+            _.each(feature.attributes, function(v, k) {
+                if(v === '' || v === null) {
+                    return;
+                }
+                newAttributes[k] = v;
+            });
+
+            newFeature.data = newFeature.properties = newFeature.attributes = newAttributes;
```

Comment:
There is really three times the same data in three different properties?

Maybe this is an example of technical debts in Mapbender. Please explain why this approach is useful, or create an issue (here or in GitLab) to discuss this strange behavior...

Reply:
@atd-schubert, yes not the best code. But OL2 and code-development-history need to do this, course somewhere is used the `data` and somewhere `attributes`. There is no copy of the data, just only linked object. And yes - confusing, but needs to been staying here.